evm/src/NttManager/NttManager.sol

```diff
@@ -291,18 +284,14 @@ contract NttManager is INttManager, NttManagerState {
                     // Since there is no standard way to query for burn fee amounts with burnable tokens,
                     // and NTT would be used on a per-token basis, implementing this functionality
                     // is left to integrating projects who may need to account for burn fees on their tokens.
-                    ERC20Burnable(token).burnFrom(msg.sender, amount);
-
-                    // query sender's token balance after transfer
-                    uint256 balanceAfter = _getTokenBalanceOf(token, msg.sender);
+                    ERC20Burnable(token).burn(amount);
 
-                    uint256 balanceDiff = balanceBefore - balanceAfter;
-                    if (balanceDiff != amount) {
-                        revert BurnAmountDifferentThanBalanceDiff(amount, balanceDiff);
+                    // tokens held by the contract after the operation should be the same as before
+                    uint256 balanceAfterBurn = _getTokenBalanceOf(token, address(this));
```

Comment:
If a user were to do this, wouldn't they just cause their own transfer to revert? I don't think it would impact any future transfers of other users?

Imo it's a good sanity check as you could have an NttManager in burning mode but the token can't be burnt (maybe it overrides `burn` and just returns true). With this check we enforce that the NttManager doesn't hold tokens when configured in burning mode

Reply:
good point!